File: tfp-did-mobile-sdk-detectid-plugin-react-native/ios/didbridge/OtpModule.m
//
//  OtpModule.m
//  DIDReact
//
//  Created by Camilo Ortegon on 10/12/22.
//

#import <Foundation/Foundation.h>

#import "React/RCTBridgeModule.h"
#import "React/RCTEventEmitter.h"

@interface RCT_EXTERN_MODULE(OtpModule,NSObject)

RCT_EXTERN_METHOD(getTokenValue:(NSString *)accountJson callback:(RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(getTokenTimeStepValue:(NSString *)accountJson callback:(RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(getChallengeQuestionOtp:(NSString *)accountJson answer:(NSString *)answer callback:(RCTResponseSenderBlock)callback)

@end

File: tfp-did-mobile-sdk-detectid-plugin-react-native/ios/didbridge/OtpModule.swift
//
//  OtpModule.swift
//  DIDReact
//
//  Created by Camilo Ortegon on 10/11/22.
//

import Foundation
import didm_sdk

@objc(OtpModule)
class OtpModule: NSObject {
  
  @objc
  static func requiresMainQueueSetup() ->Bool {
    return false;
  }
  
  @objc
  public func getTokenValue(_ accountJson: String, callback: RCTResponseSenderBlock) {
    guard let account = AccountModule.getAccount(fromJson: accountJson as String) else {
      callback([""])
      return
    }
    callback([(DetectID.sdk() as? DetectID)?.getOtpApi().getTokenValue(account) ?? ""])
  }
  
  @objc
  public func getTokenTimeStepValue(_ accountJson: String, callback: RCTResponseSenderBlock) {
    guard let account = AccountModule.getAccount(fromJson: accountJson as String) else {
      callback([""])
      return
    }
    callback(["\((DetectID.sdk() as? DetectID)?.getOtpApi().getTokenTimeStepValue(account) ?? 0)"])
  }
  
  @objc
  public func getChallengeQuestionOtp(_ accountJson: String, answer: String, callback: RCTResponseSenderBlock) {
    guard let account = AccountModule.getAccount(fromJson: accountJson as String), !answer.isEmpty else {
      callback([""])
      return
    }
    callback(["\((DetectID.sdk() as? DetectID)?.getOtpApi().getChallengeQuestionOtp(with: account, answer: answer) ?? "")"])
  }
  
}
